test/manipulation.test.js

```diff
@@ -2620,22 +2620,43 @@ describe('manipulation', function() {
         });
     });
 
-    it('fails the upsertWithWhere operation when multiple instances are ' +
-        'retrieved based on the filter criteria', function(done) {
-      Person.create([
-        {id: '2', name: 'Howie', city: 'Florida'},
-        {id: '3', name: 'Kevin', city: 'Florida'},
-      ], function(err, instance) {
-        if (err) return done(err);
-        Person.upsertWithWhere({city: 'Florida'}, {
-          id: '4', name: 'Brian',
-        }, function(err) {
-          err.message.should.equal('There are multiple instances found.' +
+    bdd.itIf(connectorCapabilities.atomicUpsertWithWhere !== true,
```

Comment:
When adding a new connector capability flag, we should pick a reasonable default so that we don't have to update all connectors with the new flag. Considering that DAO does not require connectors to implement atomic `upsertWithWhere` and provides a non-atomic implementation in such case, I think we should assume connectors don't support atomic version by default.

We need to ensure one of the `bdd.itIf` cases is executed when `connectorCapabilities.atomicUpsertWithWhere` is not set to any value (it's `undefined`). I think it will be this particular case, right?

I think the condition here could be simplified as follows:

```suggestion
    bdd.itIf(!connectorCapabilities.atomicUpsertWithWhere,
```